### PR TITLE
Fixed error message not shown when exceeding filesize

### PR DESCRIPTION
**Pull request by dobleralex**
This fixes a problem when trying to upload a file that exceeds the max allowed file size of spring. The upload was rejected, but this was not visible in the UI. It just showed the progress bar but did not do anything.

With this PR it will notify the UI properly and provide a message why the upload failed.

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java [hunk 1]
```diff
@@ -147,14 +148,15 @@ public OutputStream receiveUpload(final String fileName, final String mimeType)
      */
     @Override
     public void updateProgress(final long readBytes, final long contentLength) {
-        if (readBytes > maxSize || contentLength > maxSize) {
-            LOG.error("User tried to upload more than was allowed ({}).", maxSize);
-            interruptUploadDueToFileSizeExceeded(maxSize);
```
Comment (dobleralex) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java]: check if method still used

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java [hunk 2]
```diff
@@ -147,14 +148,15 @@ public OutputStream receiveUpload(final String fileName, final String mimeType)
...
+            publishUploadFailedAndFinishedEvent(fileUploadId);
             return;
         }
-        if (isUploadInterrupted()) {
-            // Upload interruption is delayed maybe another event is fired
-            // before
+
+        if (readBytes > maxSize || contentLength > maxSize) {
+            final String maxSizeText = SizeConversionHelper.byteValueToReadableString(maxSize);
+            LOG.error("User tried to upload more than was allowed ({}).", maxSizeText);
```
Comment (dobleralex) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java]: change log level to info or debug

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java [hunk 3]
```diff
@@ -194,11 +203,14 @@ protected void publishUploadFailedAndFinishedEvent(final FileUploadId fileUpload
         LOG.info("Upload failed for file {} due to reason: {}, exception: {}", fileUploadId, failureReason,
```
Comment (dobleralex) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java]: Check if logging the message could be done when catching the exception

Comment (dobleralex) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java]: also **consider** to change other log messages here (failed/succeeded) to debug

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java (hunk 1 above)
Comment (dobleralex) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerVaadinUpload.java]: was used in FileTransferHandlerStreamVariable; will use same approach there too; removed interruptUploadDueToFileSizeExceeded message

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java (hunk 3 above)
Comment (dobleralex) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java]: Adjusted logging levels to debug for client errors, only kept *start* and *failed*/*success* messages as info; moved log message.

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java [hunk 4]
```diff
@@ -299,7 +298,7 @@ private void streamToRepository() {
                 return Optional.ofNullable(artifactManagement.create(new ArtifactUpload(inputStream,
                         fileUploadId.getSoftwareModuleId(), filename, null, null, true, mimeType, -1)));
             } catch (final ArtifactUploadFailedException | InvalidSHA1HashException | InvalidMD5HashException e) {
-                LOG.error("Failed to transfer file to repository", e);
+                LOG.debug("Failed to transfer file to repository", e);
```
Comment (blomark) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java]: Error cases should not logged on debug level but also not on error level. I suggest to log this error case on info level.

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Pull request merged**